specs/core/0_beacon-chain.md

```diff
@@ -482,6 +485,8 @@ The following data structures are defined as [SimpleSerialize (SSZ)](https://git
 
 ```python
 {
+    'randao_reveal': 'bytes96',
```

Comment:
Why `randao_reveal` and `Eth1Data` out of header?
I understand because they are more of "operations", but is a client likely to want the randao reveals for tracking the randomness when building a header chain even if they don't want all the operations?

Reply:
> is a client likely to want the randao reveals for tracking the randomness when building a header chain even if they don't want all the operations?

I can't think of a reason off the top of my head why randao reveals would be better placed in the header. One thing to point out is that the randao reveal is just a tiny Merkle path away from the `block_body_root`, itself part of the header.

> Why `randao_reveal` and `Eth1Data` out of header?

Original motivations:

**Aesthetics**: Keeps `BeaconBlockHeader` minimal, and as you mention it may be semantically cleaner.
**Overhead**: Smaller headers mean a bit less overhead in various places (header gossiping, proposer slashings, `state.latest_block_header`).